x-pack/plugins/lists/server/services/exception_lists/put_update_exception_list_item.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsClientContract } from '@kbn/core/server';
+import type {
+  DescriptionOrUndefined,
+  EntriesArray,
+  ExceptionListItemSchema,
+  ExceptionListItemTypeOrUndefined,
+  ExpireTimeOrUndefined,
+  IdOrUndefined,
+  ItemIdOrUndefined,
+  MetaOrUndefined,
+  NameOrUndefined,
+  NamespaceType,
+  OsTypeArray,
+  TagsOrUndefined,
+  UpdateCommentsArrayOrUndefined,
+  _VersionOrUndefined,
+} from '@kbn/securitysolution-io-ts-list-types';
+import { getSavedObjectType } from '@kbn/securitysolution-list-utils';
+
+import { ExceptionListSoSchema } from '../../schemas/saved_objects';
+
+import {
+  transformSavedObjectUpdateToExceptionListItem,
+  transformUpdateCommentsToComments,
+} from './utils';
+import { getExceptionListItem } from './get_exception_list_item';
+
+interface UpdateExceptionListItemOptions {
+  id: IdOrUndefined;
+  comments: UpdateCommentsArrayOrUndefined;
+  _version: _VersionOrUndefined;
+  name: NameOrUndefined;
+  description: DescriptionOrUndefined;
+  entries: EntriesArray;
+  expireTime: ExpireTimeOrUndefined;
+  savedObjectsClient: SavedObjectsClientContract;
+  namespaceType: NamespaceType;
+  osTypes: OsTypeArray;
+  itemId: ItemIdOrUndefined;
+  meta: MetaOrUndefined;
+  user: string;
+  tags: TagsOrUndefined;
+  tieBreaker?: string;
+  type: ExceptionListItemTypeOrUndefined;
+}
+
+export const putUpdateExceptionListItem = async ({
```

Comment:
```suggestion
export const updateOverwriteExceptionListItem = async ({
```

Reply:
Maybe add a comment on this function and the exception list client function that calls it explaining the difference (and the gotcha with the regular update where undefined fields don't get reset)